src/main/java/dbaccessor/message/MessageAccessor.java

```diff
@@ -42,9 +51,21 @@ public void updateReceivedTs(Message message) {
     /**
      * Completes all DB insertions for the CreateMessage API in a single transaction.
      */
-    public boolean createMessageInTransaction(Message message, Chat chat) {
-        return spannerOperations.performReadWriteTransaction(
+    public boolean createMessageInTransaction(Message message, Chat chat, Optional<Attachment> optionalAttachment) {
+        return spannerOperations.performReadWriteTransaction (
             transactionSpannerOperations -> {
+                if (optionalAttachment.isPresent()) {
+                    Attachment attachment = optionalAttachment.get();
+                    Storage storage = StorageOptions.newBuilder().setProjectId(ApplicationVariable.GCP_PROJECT).build().getService();
+                    BlobId blobId = BlobId.of(ApplicationVariable.GCS_BUCKET, Long.toString(attachment.getAttachmentId()));
+
+                    try {
+                        storage.create(BlobInfo.newBuilder(blobId).build(), attachment.getFile().getBytes());
+                    } catch (IOException e) {
+                        return false;    
```

Comment:
So in this case, we create the attachment entry even though uploading it to cloud storage failed. Why is that so?
That doesn't seem like the right behaviour.

Reply:
No, when the `IOException` occurs, the method returns `false` and doesn't go to the statement below to `insert` the `Attachment`.